torch/utils/_sympy/value_ranges.py

```diff
@@ -111,6 +111,10 @@ def vr_is_expr(vr: ValueRanges[_T]) -> TypeGuard[ValueRanges[sympy.Expr]]:
     return not vr.is_bool
 
 
+def is_sympy_integer(value) -> TypeGuard[sympy.Integer]:
```

Comment:
Actually should be TypeIs instead of TypeGuard here, we have a stronger invariant.